### PR TITLE
systemd/emulation.go: use `systemctl --root` to enable/disable

### PR DESCRIPTION
**Pull request by valentindavid**

### CODE REVIEW

**Comment by valentindavid** [comment]
This should be supported since systemd v30. https://github.com/systemd/systemd/commit/be394c48a7cb47bf25b367cd26bcbeedda187556

I am not sure why we just did not use it.

**Comment by valentindavid** [comment]
This cleans up our code and supersedes #12489

File: systemd/emulation.go [hunk 1]
```diff
@@ -155,30 +153,7 @@ func (s *emulation) AddMountUnitFile(snapName, revision, what, where, fstype str
...
-		if err := os.MkdirAll(targetWantsDir, 0755); err != nil {
-			return "", err
-		}
-
-		enableUnitPath := filepath.Join(targetWantsDir, mountUnitName)
-		if err := os.Symlink(mu, enableUnitPath); err != nil {
-			return "", fmt.Errorf("cannot enable mount unit %s: %v", mountUnitName, err)
-		}
-	}
+	s.EnableNoReload([]string{mountUnitName})
```
Comment (pedronis) [inline systemd/emulation.go:156]: shouldn't we return any error?

File: systemd/emulation.go [hunk 2]
```diff
@@ -204,31 +179,7 @@ func (s *emulation) RemoveMountUnitFile(mountedDir string) error {
...
-			return err
-		}
-		if err := os.Remove(dir); err != nil {
-			pathErr, ok := err.(*os.PathError)
-			if !ok || pathErr.Err != syscall.ENOTEMPTY {
-				return err
-			}
-		}
-	}
+	s.DisableNoReload([]string{filepath.Base(unit)})
```
Comment (pedronis) [inline systemd/emulation.go:182]: same

**Review by pedronis — COMMENTED** [review]
thanks, this makes sense, I wondered the same yesterday looking at the other PR. Question about the errors from the functions though, as it is unusual to drop errors like this.

**Review by pedronis — COMMENTED** [review]
thank you

**Review by mvo5 — APPROVED** [review]
Woah, this is a really nice cleanup. Thanks for doing this. Fwiw, it looks like this was originally written in 4f4b71a094276dea7b22284250049be268ab09d7 and just did not consider using systemd instead.

**Review by pedronis — APPROVED** [review]
oops, actually voting

**Pull request merged**